lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,517 @@
+#include "Target/Btor/BtorToBtorIRTranslation.h"
+#include "Dialect/Btor/IR/BtorDialect.h"
+#include "Dialect/Btor/IR/BtorOps.h"
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Dialect.h"
+#include "mlir/Translation.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+
+#include <iostream>
+#include <string>
+
+using namespace mlir;
+using namespace mlir::btor;
+
+void Deserialize::parseModelLine(Btor2Line *l) {
+  reachedLines[l->id] = l;
```

Comment:
why is `reachedLines` not a vector?

Reply:
We were doubling the vector size in a way that was wasteful. I now realize we can resize it after the initial parsing of the btor file, like we were previously doing with the vector that held inits.